Src/coffeescript/cql-execution/src/elm/aggregate.coffee

```diff
@@ -36,8 +57,10 @@ module.exports.Sum = class Sum extends AggregateExpression
   exec: (ctx) ->
     arg = @source.exec(ctx)
     if typeIsArray(arg)
-      filtered =  compact(arg)
-      if filtered.length == 0 then null else filtered.reduce (x,y) -> x+y
+      arg = compact(arg)
+      filtered =  quantitiesOrArg(arg)
```

Comment:
Nevermind... Reviewing the code now, I see that the compacted version of args is also used further in the function -- so we need to original args compacted anyway.